R/object_usage_linter.R

```diff
@@ -122,9 +122,16 @@ object_usage_linter <- function(interpret_glue = TRUE, skip_with = TRUE) {
         },
         integer(1L)
       )
-
       nodes <- unclass(lintable_symbols)[matched_symbol]
-      nodes[is.na(matched_symbol)] <- list(fun_assignment)
+
+      # fallback to line based matching if no symbol is found
+      nodes[is.na(matched_symbol)] <- lapply(which(is.na(matched_symbol)), function(i) {
+        line_based_match <- xml2::xml_find_first(
+          fun_assignment,
+          glue::glue("descendant::expr[@line1 = {res$line1[i]} and @line2 = {res$line2[i]}]")
```

Comment:
Not sure if `find_first` will always find the outermost expression, i.e. the one covering the complete line range, but no additional lines.
I'm 90% sure though, because all other exprs will be descendants of that specific one.

Reply:
Hmm, something like is `xml_find_first()` doing a greedy search under `descendant::expr`? FWIW here's a quick example:

```r
xml = read_xml('<x><a><a><a><a></a></a></a></a></x>')
xml |> xml_find_first('//x/descendant::a') |> as.character() |> writeLines()
<a>
  <a>
    <a>
      <a/>
    </a>
  </a>
</a>
```

does appear to find the outermost one. Not sure if that's generalizable, or if it's specifically what you had in mind.